vslm/global_object_manager.go

```diff
@@ -0,0 +1,659 @@
+	}
+
+	res, err := methods.VslmRevertVStorageObject_Task(ctx, this.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewTask(this.c, res.Returnval), nil
+}
+
+func (this *GlobalObjectManager) QueryChangedDiskAreas(ctx context.Context, id vim.ID, snapshotId vim.ID, startOffset int64,
+	changeId string) (*vim.DiskChangeInfo, error) {
+	req := types.VslmQueryChangedDiskAreas{
+		This:        this.Reference(),
+		Id:          id,
+		SnapshotId:  snapshotId,
+		StartOffset: startOffset,
+	}
+
+	res, err := methods.VslmQueryChangedDiskAreas(ctx, this.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return &res.Returnval, nil
+}
+
+func (this *GlobalObjectManager) queryGlobalCatalogSyncStatus(ct context.Context) ([]types.VslmDatastoreSyncStatus, error) {
+	req := types.VslmQueryGlobalCatalogSyncStatus{
+		This: this.Reference(),
+	}
+
+	res, err := methods.VslmQueryGlobalCatalogSyncStatus(ct, this.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return res.Returnval, nil
+}
+
+func (this *GlobalObjectManager) queryGlobalCatalogSyncStatusForDatastore(ct context.Context, datastoreURL string) (
+	*types.VslmDatastoreSyncStatus, error) {
+	req := types.VslmQueryGlobalCatalogSyncStatusForDatastore{
+		This:         this.Reference(),
+		DatastoreURL: datastoreURL,
+	}
+
+	res, err := methods.VslmQueryGlobalCatalogSyncStatusForDatastore(ct, this.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return res.Returnval, nil
+}
+
+func (this *GlobalObjectManager) UpdateMetadata(ctx context.Context, id vim.ID, metadata []vim.KeyValue,
+	deleteKeys []string) (*Task, error) {
+	req := &types.VslmUpdateVStorageObjectMetadata_Task{
+		This:       this.Reference(),
+		Id:         id,
+		Metadata:   metadata,
+		DeleteKeys: deleteKeys,
+	}
+
+	res, err := methods.VslmUpdateVStorageObjectMetadata_Task(ctx, this.c, req)
+	if err != nil {
+		return nil, err
+	}
+	return NewTask(this.c, res.Returnval), nil
+}
+
+func (this *GlobalObjectManager) RetrieveMetadata(ct context.Context, id vim.ID, snapshotID *vim.ID, prefix string) (
+	[]vim.KeyValue, error) {
+	req := types.VslmRetrieveVStorageObjectMetadata{
+		This:       this.Reference(),
+		Id:         id,
+		SnapshotId: snapshotID,
+		Prefix:     prefix,
+	}
+
+	res, err := methods.VslmRetrieveVStorageObjectMetadata(ct, this.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return res.Returnval, nil
+}
+
+func (this *GlobalObjectManager) RetrieveMetadataValue(ct context.Context, id vim.ID, snapshotID *vim.ID, key string) (
+	string, error) {
+	req := types.VslmRetrieveVStorageObjectMetadataValue{
+		This:       this.Reference(),
+		Id:         id,
+		SnapshotId: snapshotID,
+		Key:        key,
+	}
+
+	res, err := methods.VslmRetrieveVStorageObjectMetadataValue(ct, this.c, &req)
+	if err != nil {
+		return "", err
+	}
+
+	return res.Returnval, nil
+}
+
+func (this *GlobalObjectManager) retrieveObjects(ct context.Context, ids []vim.ID) ([]types.VslmVsoVStorageObjectResult,
```

Comment:
Should this method be exported along with the one below?

Reply:
Yup